tools/chapel-py/src/method-tables/core-methods.h

```diff
@@ -0,0 +1,116 @@
+         void(std::vector<std::string>, std::vector<std::string>),
+
+         auto& paths = std::get<0>(args);
+         auto& filenames = std::get<1>(args);
+         parsing::setupModuleSearchPaths(node, false, false, paths, filenames))
+  METHOD(Context, is_bundled_path, "Check if the given file path is within the bundled (built-in) Chapel files",
+         bool(chpl::UniqueString),
+
+         auto pathUS = std::get<0>(args);
+         return parsing::filePathIsInInternalModule(node, pathUS) ||
+                parsing::filePathIsInStandardModule(node, pathUS) ||
+                parsing::filePathIsInBundledModule(node, pathUS))
+  METHOD(Context, advance_to_next_revision, "Advance the context to the next revision",
+         void(bool),
+
+         auto prepareToGc = std::get<0>(args);
+         node->advanceToNextRevision(prepareToGc))
+CLASS_END(Context)
+
+CLASS_BEGIN(Location)
+  PLAIN_GETTER(Location, start, "Get the line-column pair where this Location starts",
+               LineColumnPair, return std::make_tuple(node->firstLine(), node->firstColumn()))
+  PLAIN_GETTER(Location, end, "Get the line-column pair where this Location ends",
+               LineColumnPair, return std::make_tuple(node->lastLine(), node->lastColumn()))
+  PLAIN_GETTER(Location, path, "Get the file path of this Location",
+               chpl::UniqueString, return node->path())
+CLASS_END(Location)
+
+CLASS_BEGIN(Scope)
+  PLAIN_GETTER(Scope, used_imported_modules, "Get the modules that were used or imported in this scope",
+               std::vector<const chpl::uast::AstNode*>,
+
+               auto& moduleIds = resolution::findUsedImportedModules(context, node);
+               std::set<ID> reportedIds;
+               std::vector<const chpl::uast::AstNode*> toReturn;
+               for (size_t i = 0; i < moduleIds.size(); i++) {
+                 auto& id = moduleIds[i];
+                 if (!reportedIds.insert(id).second) continue;
+                 toReturn.push_back(parsing::idToAst(context, id));
+               }
+               return toReturn)
+CLASS_END(Scope)
+
+CLASS_BEGIN(Error)
+  PLAIN_GETTER(Error, location, "Get the location at which this error occurred",
+               chpl::Location, return node->location(context))
+  PLAIN_GETTER(Error, message, "Retrieve the contents of this error message",
+               std::string, return node->message())
+  PLAIN_GETTER(Error, kind, "Retrieve the kind ('error', 'warning') of this type of error",
+               const char*, return chpl::ErrorBase::getKindName(node->kind()))
+  PLAIN_GETTER(Error, type, "Retrieve the unique name of this type of error",
+               const char*, return chpl::ErrorBase::getTypeName(node->type()))
+CLASS_END(Error)
+
+CLASS_BEGIN(ErrorManager)
+  PLAIN_GETTER(ErrorManager, __enter__, "The context manager 'enter' method for this ErrorManager object",
+               PyObject*,
+
+               (void) node;
```

Comment:
I am guessing the `(void) node;` here is to suppress unused variables warning? I would prefer either using `std::ignore` directly here, or putting a `std::ignore` into the macro def that instantiates this x-macro

Reply:
I think it should go here because in general it ought to be a warning to ignore `node`.